### ISSUE TITLE
Running "make run-linux" gives "KeyError: 'mbgl.gyp:headless-glx#target'"

### ISSUE THREAD

**Issue by springmeyer** [mapbox/mapbox-gl-native#1206]
Hitting this build error I've not seen before:

```
$ make run-linux
deps/run_gyp linux/mapboxgl-app.gyp -Dhost=osx -Iconfig/osx.gypi -Dinstall_prefix=/usr/local -Dheadless_lib=glx -Dplatform_lib=linux -Dasset_lib=fs -Dhttp_lib=curl -Dcache_lib=sqlite --depth=. -Goutput_dir=. --generator-output=./build/osx -f make
Traceback (most recent call last):
  File "deps/run_gyp", line 8, in <module>
    sys.exit(gyp.script_main())
  File "deps/gyp/pylib/gyp/__init__.py", line 534, in script_main
    return main(sys.argv[1:])
  File "deps/gyp/pylib/gyp/__init__.py", line 527, in main
    return gyp_main(args)
  File "deps/gyp/pylib/gyp/__init__.py", line 503, in gyp_main
    options.circular_check)
  File "deps/gyp/pylib/gyp/__init__.py", line 129, in Load
    params['parallel'], params['root_targets'])
  File "deps/gyp/pylib/gyp/input.py", line 2759, in Load
    RemoveLinkDependenciesFromNoneTargets(targets)
  File "deps/gyp/pylib/gyp/input.py", line 1483, in RemoveLinkDependenciesFromNoneTargets
    if targets[t].get('variables', {}).get('link_dependency', 0):
KeyError: 'mbgl.gyp:headless-glx#target'
make: *** [Makefile/linux] Error 1
```

**Comment by springmeyer** [comment]
Looks like a potential fix is:

```
diff --git a/mbgl.gyp b/mbgl.gyp
index c5ede28..b51502b 100644
--- a/mbgl.gyp
+++ b/mbgl.gyp
@@ -10,6 +10,7 @@
     './gyp/none.gypi',
   ],
   'conditions': [
+    ['headless_lib == "glx" and host == "osx"', { 'includes': [ './gyp/headless-glx.gypi' ] } ],
     ['headless_lib == "cgl" and host == "osx"', { 'includes': [ './gyp/headless-cgl.gypi' ] } ],
     ['headless_lib == "glx" and host == "linux"', { 'includes': [ './gyp/headless-glx.gypi' ] } ],
     ['platform_lib == "osx" and host == "osx"', { 'includes': [ './gyp/platform-osx.gypi' ] } ],
```

This raises the question: is it still intended that you can test the linux app on OS X? (/cc @kkaefer). For context: the reason I'm trying to run the linux app is to workaround #1205.

**Comment by kkaefer** [comment]
Nope, the "fix" is to use `make linux HEADLESS=cgl`

**Comment by kkaefer** [comment]
Fixed by #1207.
